testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy

```diff
@@ -167,7 +249,11 @@ abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
         def coreLibs = contentsDir.file("lib").listFiles().findAll {
             it.name.startsWith("gradle-") && !it.name.startsWith("gradle-api-metadata") && !it.name.startsWith("gradle-kotlin-dsl")
         }
+
+        coreLibs.each { assert coreLibsContains(it.name) }
         assert coreLibs.size() == coreLibJarsCount
```

Comment:
:x: The `coreLibJarsCount` should go, and we should use the explicit list size instead in places that still need the count.

Reply:
using the list size in `getCoreLibJarsCount`